.github/workflows/dependabot.yml

```diff
@@ -0,0 +1,56 @@
+name: Dependabot Workflow
+
+# This workflow commits to Dependabot branches to ensure that the corresponding
+# PRs can satisfy all status checks.
+
+# WARNING: Combining pull_request_target workflow trigger with an explicit
+# checkout of an untrusted PR is a dangerous practice that may lead to
+# repository compromise.
+# See https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
+# To prevent repository compromise, the workflow jobs must only execute on PRs
+# opened by Dependabot and which are labelled correctly (note that these two
+# checks are somewhat redundant since labelling PRs require write access to the
+# repository).
+# An alternative is to use the "two-workflow method" (see
+# https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/automating-dependabot-with-github-actions#handling-push-events),
+# but that is more tedious to configure and should not be required here.
+
+on:
+  pull_request_target:
+    types: [labeled, synchronize]
+
+permissions:
+  contents: write
+
+jobs:
+  # This job ensures that "go mod tidy" is run for all Go modules included in
+  # this repository.
+  tidy:
+    name: Go tidiness for Dependabot PR
+    # 'dependencies' and 'go' are the default labels used by Dependabot when updating Go dependencies
+    if: ${{ github.actor == 'dependabot[bot]' && contains(github.event.pull_request.labels.*.name, 'dependencies') && contains(github.event.pull_request.labels.*.name, 'go') }}
+    runs-on: [ubuntu-latest]
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        # Check out the pull request HEAD
+        ref: ${{ github.event.pull_request.head.sha }}
+        token: ${{ secrets.ANTREA_BOT_WRITE_PAT }}
```

Comment:
do you know whether theia has this secret? I have no access to repository secrets so I cannot check. Have you already checked it?

Reply:
I'm a bit confused. Do you mean you created a secret in your forked theia repo? If we would like to merge this PR, shouldn't we create a secret for this theia repo?

Reply:
I see the documentation: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/configuring-access-to-private-registries-for-dependabot#about-encrypted-secrets-for-dependabot

> When you add a secret at the organization level, you can specify which repositories can access the secret.
Do we know if this secret is set at the organization level and theia repo has access to the secret?

Reply:
In antrea PRs, we can find the Dependabot workflow, e.g.
https://github.com/antrea-io/antrea/actions/runs/4556122564/jobs/8036076618?pr=4768

but in this PR, we didn't see it showing in the CI checks?

Reply:
Sorry for the confusion, what I meant is for theia we right now do not have a PAT, we can use this for right now as a placeholder, but it won’t do anything as it is only present for Antrea, once a new PAT becomes available for Theia, we will add that in ANTREA_BOT_WRITE_PAT’s place.